### PR TITLE
XDR-3534: Add data connector for cybereason malops

### PR DESCRIPTION
**Pull request by deepanshumarwah**

### CODE REVIEW

File: DataConnectors/Cybereason/cybereason-pull/readme.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+The `TimerTrigger` makes it incredibly easy to have your functions executed on a schedule. This sample demonstrates a simple use case of calling your function every 5 minutes.
+
+## How it works
+
+For a `TimerTrigger` to work, you provide a schedule in the form of a [cron expression](https://en.wikipedia.org/wiki/Cron#CRON_expression)(See the link for full details). A cron expression is a string with 6 separate expressions which represent a given schedule via patterns. The pattern we use to represent every 5 minutes is `0 */5 * * * *`. This, in plain text, means: "When seconds is equal to 0, minutes is divisible by 5, for any hour, day of the month, month, day of the week, or year".
+
+## Learn more
+
+<TODO> Documentation
```
Comment (arledesma) [inline DataConnectors/Cybereason/cybereason-pull/readme.md:11]: Is this needed?


```suggestion
```

File: DataConnectors/Cybereason/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+## Description
+This log source uses logic app workflow with log analytics send data plugin. This cybereason connector pulls in data from malops using 2 apis `detection/inbox` and `detection/details`. For additional logs, you can modify the workflow with additional apis. For more information refer to https://nest.cybereason.com/documentation/api-documentation/all-versions/
+
+## Prerequisites
+1. Cybereason username, password and hostname.
+3. Log analytics workspace ID and shared Key
+
+
+## Deploy the Azure Function using ARM template
+[![Deploy to Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FDataConnectors%2FCybereason%2Fazuredeploy_connector_cybereasonjson)
```
Comment (arledesma) [inline DataConnectors/Cybereason/README.md:10]: It looks like there is a missing `.` 

```suggestion
[![Deploy to Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FDataConnectors%2FCybereason%2Fazuredeploy_connector_cybereason.json)
```

**Review by arledesma — CHANGES_REQUESTED** [review]
Mainly nitpicking on style.

The only thing that must be changed is the url in the readme.

File: DataConnectors/Gitlab/README.md [hunk 3]
```diff
@@ -11,7 +11,7 @@ This log source uses logic app workflow with log analytics send data plugin. Thi
 
 
 ## Deploy the Logic App template
-<!-- [![Deploy to Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/<Tobeadded>) -->
+[![Deploy to Azure](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FDataConnectors%2FGitlab%2Fazuredeploy.json)
```
[the remark this answers is not in the document]
Comment (deepanshumarwah) [reply DataConnectors/Gitlab/README.md:14]: Not really but the url wasn't updated previously. So I added this here. Also, the url would only work once it is merged to their public repo.

File: DataConnectors/Cybereason/cybereason-pull/__init__.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        logging.info("The last time point is: {}".format(past_time))
+    else:
+        logging.info("There is no last time point, trying to get events for last hour.")      
+        past_time = current_time - datetime.timedelta(minutes=60)
+    state.post(current_time)    
+    return (int(past_time.timestamp() * 1000.0) , int(current_time.timestamp() * 1000.0))
+
+def build_signature(customer_id, shared_key, date, content_length, method, content_type, resource):
+    x_headers = 'x-ms-date:' + date
+    string_to_hash = method + "\n" + str(content_length) + "\n" + content_type + "\n" + x_headers + "\n" + resource
```
[the remark this answers is not in the document]
Comment (deepanshumarwah) [reply DataConnectors/Cybereason/cybereason-pull/__init__.py:109]: This is concise. I like it.

File: DataConnectors/Cybereason/cybereason-pull/__init__.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    exit(1)
+  return session
+
+
+def get_detections(start_time, end_time):
+  malops_url = base_url + malops_endpoint
+  query = {
+    "startTime":start_time,
+    "endTime":end_time
+  }
```
[the remark this answers is not in the document]
Comment (deepanshumarwah) [reply DataConnectors/Cybereason/cybereason-pull/__init__.py]: Let me take a look into that

File: DataConnectors/Cybereason/cybereason-pull/readme.md (hunk 1 above)
Comment (deepanshumarwah) [reply DataConnectors/Cybereason/cybereason-pull/readme.md]: Removed this

File: DataConnectors/Cybereason/cybereason-pull/__init__.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (deepanshumarwah) [reply DataConnectors/Cybereason/cybereason-pull/__init__.py]: I added a utility called Trunk. It has linter for all type of files.

[the remark this answers is not in the document]
Comment (arledesma) [reply DataConnectors/Cybereason/cybereason-pull/__init__.py]: I like trunk but ultimately ended up uninstalling it. It's a cool concept and will definitely work for this use-case.

**Review by arledesma — APPROVED** [review]
:thumbsup:

File: __azurite_db_table__.json [hunk 6]
```diff
@@ -0,0 +1,114 @@
+{
+  "filename": "/Users/deepanshumarwah/projects/quantum-sec/Azure-Sentinel/__azurite_db_table__.json",
```
Comment (arledesma) [inline __azurite_db_table__.json:2]: Ouch, didn't notice this before

**Review by arledesma — CHANGES_REQUESTED** [review]
Maybe one more change

File: __azurite_db_table__.json (hunk 6 above)
Comment (deepanshumarwah) [reply __azurite_db_table__.json:2]: oh shoot

Comment (deepanshumarwah) [reply __azurite_db_table__.json]: removed this and also there was a space issue in app kind.

File: DataConnectors/Cybereason/azuredeploy_connector_cybereason.json [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+    "StorageSuffix": "[environment().suffixes.storage]",
+    "LogAnaltyicsUri": "[replace(environment().portal, 'https://portal', concat('https://', toLower(parameters('WorkspaceID')), '.ods.opinsights'))]"
+  },
+  "resources": [
+    {
+      "name": "[variables('FunctionName')]",
+      "type": "Microsoft.Web/sites",
+      "apiVersion": "2020-12-01",
+      "location": "[resourceGroup().location]",
+      "kind": "functionapp,linux",
```
Comment (arledesma) [inline DataConnectors/Cybereason/azuredeploy_connector_cybereason.json:43]: I would not have noticed that extra space. 👍

**Review by arledesma — APPROVED** [review]

**Pull request merged**